client/src/components/RecipeCard.tsx

```
@@ -0,0 +1,171 @@
+  /*   const userId = user?.userId; */
+  useEffect(() => {
+    async function fetchIsLikes() {
+      const req = {
+        method: 'get',
+        headers: {
+          'Content-Type': 'application/json',
+        },
+      };
+      const response = await fetch(`/api/likes/${userId}/${recipeId}`, req);
+      if (!response.ok) throw new Error('Network response not ok.');
+      const data = await response.json();
+      if (data === 'dislike') {
+        setIsLikes(false);
+      } else {
+        setIsLikes(true);
+      }
+    }
+    async function fetchLikesCount() {
+      try {
+        if (!recipeId) throw new Error('recipeId required.');
+        const response = await fetch(`/api/likes/${recipeId}`);
+        const data = await response.json();
+        const count = data.count as number;
+        if (!count) return setLikesCount(0);
+        setLikesCount(count);
+      } catch (error) {
+        console.log(error);
+        throw new Error('fetch likes count failed');
+      }
+    }
+    if (user) {
+      fetchIsLikes();
+    }
+
+    fetchLikesCount();
+  }, [isLikes, likesCount]);
+
+  function handleClick() {
+    navigate('/details', { state: recipe });
+  }
+
+  async function handleFaClickData(recipeId: string, userId: string) {
+    if (user) {
+      try {
+        const req = {
+          method: 'POST',
+          headers: {
+            'Content-Type': 'application/json',
+            Authorization: `Bearer ${readToken()}`,
+          },
+          body: JSON.stringify({ recipeId, userId }),
+        };
+        const response = await fetch('/api/likes', req);
+        if (!response.ok) throw new Error('Network response not ok.');
+        const data = await response.json();
+        if (data === 'inserted') {
+          setIsLikes(true);
+        } else {
+          setIsLikes(false);
+        }
+      } catch (error) {
+        console.error(error);
+        throw new Error('faHeart clicks failed.');
```

Same comment here about throwing an exception. And about "fail fast" error handling.